packages/hdwallet-tally/src/tally.ts

```diff
@@ -0,0 +1,340 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import * as eth from "./ethereum";
+import _ from "lodash";
+import detectEthereumProvider from "@metamask/detect-provider";
+
+class TallyTransport extends core.Transport {
+  public async getDeviceID() {
+    return "tally:0";
+  }
+
+  public call(...args: any[]): Promise<any> {
+    return Promise.resolve();
+  }
+}
+
+export function isTally(wallet: core.HDWallet): wallet is TallyHDWallet {
+  return _.isObject(wallet) && (wallet as any)._isTally;
+}
+
+type HasNonTrivialConstructor<T> = T extends { new (): any } ? never : T;
+
+export class TallyHDWallet implements core.HDWallet, core.ETHWallet {
+  readonly _supportsETH = true;
+  readonly _supportsETHInfo = true;
+  readonly _supportsBTCInfo = false;
+  readonly _supportsBTC = false;
+  readonly _supportsCosmosInfo = false;
+  readonly _supportsCosmos = false;
+  readonly _supportsOsmosisInfo = false;
+  readonly _supportsOsmosis = false;
+  readonly _supportsBinanceInfo = false;
+  readonly _supportsBinance = false;
+  readonly _supportsDebugLink = false;
+  readonly _isPortis = false;
+  readonly _isTally = true;
+  readonly _supportsRippleInfo = false;
+  readonly _supportsRipple = false;
+  readonly _supportsEosInfo = false;
+  readonly _supportsEos = false;
+  readonly _supportsFioInfo = false;
+  readonly _supportsFio = false;
+  readonly _supportsThorchainInfo = false;
+  readonly _supportsThorchain = false;
+  readonly _supportsSecretInfo = false;
+  readonly _supportsSecret = false;
+  readonly _supportsKava = false;
+  readonly _supportsKavaInfo = false;
+  readonly _supportsTerra = false;
+  readonly _supportsTerraInfo = false;
+
+  transport: core.Transport = new TallyTransport(new core.Keyring());
+  info: TallyHDWalletInfo & core.HDWalletInfo;
+  ethAddress?: string | null;
+  provider: any;
+
+  constructor() {
+    this.info = new TallyHDWalletInfo();
+  }
+
+  async getFeatures(): Promise<Record<string, any>> {
+    return {};
+  }
+
+  public async isLocked(): Promise<boolean> {
+    return !this.provider.tally.isUnlocked();
+  }
+
+  public getVendor(): string {
+    return "Tally";
+  }
+
+  public getModel(): Promise<string> {
+    return Promise.resolve("Tally");
+  }
+
+  public getLabel(): Promise<string> {
+    return Promise.resolve("Tally");
+  }
+
+  public async initialize(): Promise<any> {
+    try {
+      this.provider = await detectEthereumProvider({ mustBeMetaMask: false, silent: false, timeout: 3000 });
+    } catch (e) {
+      console.error(e);
+    }
+
+    return Promise.resolve();
+  }
```

Comment:
as in #431
```suggestion
  public async initialize(): Promise<boolean> {
    try {
      this.provider = await detectEthereumProvider({ mustBeMetaMask: false, silent: false, timeout: 3000 });
      return true;
    } catch (e) {
      console.error(e);
      return false;
    }
  }
```

Reply:
Fixed in [0b0e667](https://github.com/shapeshift/hdwallet/pull/506/commits/0b0e667653a33fb7fb9ddd4905b81979b6cb1fa8)